### ISSUE TITLE
Unable to reset vegacapsule network on Jenkins during a system test run

### ISSUE THREAD

**Issue by jgsbennett** [vegaprotocol/vegacapsule#298]
The new protocol upgrade tests (such as those included on [this PR](https://github.com/vegaprotocol/system-tests/pull/1322)) need to reset the vegacapsule network between tests, to revert to a network running the "current" version of vega. Unfortunately, when run on Jenkins, the network fails to reset, with some of the clef nodes reporting errors connecting to clef.

Errors are like this:
```
[2022-10-17T09:54:49.428Z] 
[2022-10-17T09:54:49.428Z] Error: failed to generate network: failed to initiate Vega node id 4 for node set validators-clef: failed to import "ethereum" wallet: failed to execute binary /jenkins/workspace/common/system-tests-wrapper/tests/vega [nodewallet --home /jenkins/workspace/common/system-tests-wrapper/tests/testnet/vega/node4 --passphrase-file /jenkins/workspace/common/system-tests-wrapper/tests/testnet/vega/node4/node-vega-wallet-pass.txt import --output json --chain ethereum --ethereum-clef-account 0xc5bf831ddf098fe218958d74676c6ffc64baa970 --ethereum-clef-address http://localhost:8554/] with error: couldn't import Ethereum node wallet: couldn't initialise Ethereum Clef node wallet: account not found: failed to list accounts: failed to call client: Post "http://localhost:8554/": dial tcp 127.0.0.1:8554: connect: connection refused
[2022-10-17T09:54:49.428Z] : exit status 255
```
... despite vegacapsule having originally reported success starting the clef node.

See this job:
https://jenkins.ops.vega.xyz/job/common/job/system-tests-wrapper/18297/
(Once this run completes, I'll grab some more logs and paste onto this ticket.)

**Comment by daniel1302** [comment]
Parameters to trigger a job:

```
library (
    identifier: "vega-shared-library@pup_changes",
    changelog: false,
)

capsuleSystemTests([
    hooks: [
        preNetworkStop: [
            'Wait': {
                sleep 1800
            }    
        ]
    ]
], [
    SYSTEM_TESTS_BRANCH: "protocol_upgrade",
    CAPSULE_CONFIG: "capsule_config_network_infra.hcl",
    SYSTEM_TESTS_TEST_MARK: "protocol_upgrade",
    
])
```

**Comment by jgsbennett** [comment]
Just for record-keeping, the vegacapsule world has mysteriously changed in the couple of weeks since this issue first appeared:
@karlem @daniel1302 and I looked at this today, and the behaviour has changed. The network reset appears to succeed, and instead the next test hangs trying to connect to the network in one of the test worker threads, even though the network reported being started correctly. Daniel logged into the Jenkins agent and confirmed that the network appeared to be up and running, so it's back over to me to fix the case where the test hangs so that we can see _where it is actually trying to connect to_, and the bat back this way if it's not immediately obvious how to remedy that.

**Comment by gordsport** [comment]
@jgsbennett - I have removed this from the backlog and into triage. I believe that you are unblocked. Should this happening again if the details in this issue still remain we can add it back in, alternatively a new issue with updated details should be created.

**Comment by gordsport** [comment]
Expected this will be resolved with:
https://github.com/vegaprotocol/docker/pull/65

**Comment by gordsport** [comment]
@jgsbennett 

Can you please re-test this now the above has been merged

**Comment by gordsport** [comment]
@jgsbennett - please can you let me know if this is still required?

**Comment by jgsbennett** [comment]
This would be a nice thing to add to vegacapsule at some point. Were it added to vegacapsule, I would likely modify the tests to use it, rather than our current solution which is to destroy the current network and bootstrap fresh again. There is an argument that users of vegacapsule might like this feature themselves, meaning this is either a feature request or enhancement for the benefit of other vegacapsule users (who can also do the workaround we have), which has no current impact on the tests.

**Comment by jgsbennett** [comment]
Revising upwards how useful this ticket is, slightly. Many of the QA team (and likely the devs who are now also contributing tests) run the tests within PyCharm. This may mean that they started the network from _outside_ of PyCharm, or at least, not from the same terminal. Currently the existing test solution relies on having been run from the same environment as that which started the test network, and hence, it can work out which hcl was used to start the network and use that again. If we don't start the test from the same environment, this doesn't work.

It's _possible_ we could add a better test workaround to attempt to detect the existing network and keep track on that information better for the sake of this utility function, but it feels better for VC to be able to be asked to reset the original network, or at a minimum, explicitly tell us which hcl file was used to start the current network.

**Comment by jgsbennett** [comment]
I've realised that this issue is not actually an issue any more, and got mixed up with https://github.com/vegaprotocol/vegacapsule/issues/268 , which is the remaining work we still need done.
